pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -860,39 +873,49 @@ def append_tk_command_to_qiskit(
         for i, a in enumerate(args[:width]):
             if a.reg_name != regname:
                 raise NotImplementedError("Conditions can only use a single register")
-        instruction = append_tk_command_to_qiskit(
-            op.op,
-            args[width:],
-            qcirc,
-            qregmap,
-            cregmap,
-            symb_map,
-            range_preds,
-        )
         if len(cregmap[regname]) == width:
             for i, a in enumerate(args[:width]):
                 if a.index != [i]:
                     raise NotImplementedError(
                         """Conditions must be an entire register in\
  order or only one bit of one register"""
                     )
-
-            instruction.c_if(cregmap[regname], value)
+            condition = (cregmap[regname], value)
         elif width == 1:
-            instruction.c_if(cregmap[regname][args[0].index[0]], value)
+            condition = (cregmap[regname][args[0].index[0]], value)
         else:
             raise NotImplementedError(
                 """Conditions must be an entire register in\
 order or only one bit of one register"""
             )
+        instruction = append_tk_command_to_qiskit(
+            op.op,
+            args[width:],
+            qcirc,
+            qregmap,
+            cregmap,
+            symb_map,
+            range_preds,
+            condition=condition,
+        )
 
         return instruction
     # normal gates
     qargs = [qregmap[q.reg_name][q.index[0]] for q in args]
     if optype == OpType.CnX:
-        return qcirc.mcx(qargs[:-1], qargs[-1])
+        if condition is None:
+            qcirc.mcx(qargs[:-1], qargs[-1])
+        else:
+            with qcirc.if_test(condition):
+                qcirc.mcx(qargs[:-1], qargs[-1])
+        return qcirc
     if optype == OpType.CnY:
-        return qcirc.append(qiskit_gates.YGate().control(len(qargs) - 1), qargs)
+        if condition is None:
+            qcirc.append(qiskit_gates.YGate().control(len(qargs) - 1), qargs)
+        else:
+            with qcirc.if_test(condition):
+                qcirc.append(qiskit_gates.YGate().control(len(qargs) - 1), qargs)
+        return qcirc
     if optype == OpType.CnZ:
         if len(qargs) == 2:
```

Comment:
TODO check that I didn't mess up the handling of controlled-Z gates in the merge -> 23289c8f4b9e7e596e6753a2eb4c528b6e3223ae